src/irmin-pack/content_addressable_intf.ml

```diff
@@ -83,27 +75,57 @@ module type S = sig
   val clear_keep_generation : 'a t -> unit Lwt.t
 end
 
+module type Indexed_S = sig
```

Comment:
I'm not sure if this naming matches the pattern recently adopted by irmin.

Maybe`Indexed.S` and `Indexed.Maker` instead of `Indexed_S` and `Indexed_maker` would fit better.